find_attitude/web_find_attitude/views.py

```diff
@@ -1,14 +1,21 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import os
-from django.shortcuts import render
+from logging import CRITICAL
 
+from django.shortcuts import render
 from kadi import __version__
-from find_attitude.find_attitude import get_stars_from_text, find_attitude_solutions, logger
-from logging import CRITICAL
+
+from find_attitude.find_attitude import (
+    find_attitude_solutions,
+    get_stars_from_text,
+    logger,
+)
 
 # Only emit critical messages
 logger.level = CRITICAL
-for hdlr in logger.handlers:
+# FIXME: this looks like a bug in the original code. Either get rid of the loop or
+# set levels for all handlers along with the logger.
```

Comment:
`logging` never fails to surprise and confuse me. E.g. if `logger.level = CRITICAL` does not work then `logging` should not allow that. I agree the code here is not ideal (hence the comment), but it *does* actually work.
```
In [1]: import os
   ...: from logging import CRITICAL
   ...: 
   ...: from django.shortcuts import render
   ...: from kadi import __version__
   ...: 
   ...: from find_attitude.find_attitude import (
   ...:     find_attitude_solutions,
   ...:     get_stars_from_text,
   ...:     logger,
   ...: )
   ...: 

In [2]: logger.info("info")
2024-03-28 10:26:35,922 info

In [3]: # Only emit critical messages
   ...: logger.level = CRITICAL
   ...: # FIXME: this looks like a bug in the original code. Either get rid of the loop or
   ...: # set levels for all handlers along with the logger.
   ...: for _hdlr in logger.handlers:
   ...:     logger.setLevel(CRITICAL)
   ...: 

In [4]: logger.info("info")

In [5]: logger.critical("critical")
2024-03-28 10:27:05,944 critical
```
This PR is not intended for bug fixes. I'll open a new issue for this.

Reply:
See #28

Reply:
I agree that logging has all these weird things. I tested before I wrote above.

The reason why your code works is because `logger.setLevel(CRITICAL)` sets it at the logger level and that seems to be enough in this case (I was surprised about that, given all the logging weirdness).

```
In [1]: import logging
   ...: logging.basicConfig(level="DEBUG")
   ...: logger = logging.getLogger('bla')
   ...: logger.info("hey")
   ...: logger.level = logging.CRITICAL
   ...: logger.info("hey")
   ...: 
INFO:bla:hey
INFO:bla:hey

```

Reply:
Yes, if `logger.level = level` does not work then it should raise an exception. And sounding like a broken record, `logging` should never have included a `basicConfig` that sets the root logger and is recommended as the right thing to do in  all the docs. I'm not sure if you edited your output, but here is what I got:
```
In [1]: In [1]: import logging
   ...:    ...: logging.basicConfig(level="DEBUG")
   ...:    ...: logger = logging.getLogger('bla')

DEBUG:asyncio:Using selector: KqueueSelector
In [2]: 
```
NOOOO!!! I do NOT want to see debug output from asyncio. NO!